lwip/lwip_cpp/LightweightIpOverEthernet.cpp

```diff
@@ -185,6 +185,7 @@ namespace services
         netif_set_up(&netInterface);
         netif_set_default(&netInterface);
 
+        really_assert((config.hostName.size() < config.hostName.max_size() && config.hostName.Storage()[config.hostName.size()] == '\0') || (config.hostName.Storage()[config.hostName.size() - 1] == '\0'));
```

Comment:
Great to add a check on 0-termination, but I think the 0 should in all cases be part of the string, not just outside it.
I guess you wanted to support hostnames from string literals? A BoundedConstString can be constructed from such a name in this way:

    const char hostname[] = "host.com";
    infra::BoundedConstString(hostname, sizeof(hostname));

With that in mind, would you agree to simplifying the check to this?
```suggestion
        really_assert(config.hostName.Storage()[config.hostName.size() - 1] == '\0');
```

Reply:
Thanks for the suggestion, I agree that it is better and makes it more explicit that the '\0' must be part of the BoundedString. I was constructing the BoundedString with

infra::BoundedString::WithStorage<15> hostname{"eth-lwip-cpusim"};

which does apparently not add a '\0' in memory.

Reply:
I looked at the needed changes but sadly it would introduce to many side effects, for example with length/size calculation and concatenation/appending so after discussion with Fabian we think it is not worth the effort.